x-pack/plugins/observability/server/services/slo/slo_repository.ts

```diff
@@ -117,26 +128,32 @@ export class KibanaSavedObjectsSLORepository implements SLORepository {
       total: response.total,
       perPage: response.per_page,
       page: response.page,
-      results: response.saved_objects.map((slo) => toSLO(slo.attributes)),
+      results: response.saved_objects
+        .map((savedObject) => this.toSLO(savedObject.attributes))
+        .filter((slo) => slo !== undefined) as SLO[],
```

Comment:
i have concern regarding that, if user has an invalid SLO, it will never appear in the list and they will not know to delete it.

Reply:
Discussed offline, the purpose of this fix is not to find out about invalid SLO but to prevent failures of any read when one SLO is invalid. Which can happen (and already happened) when upgrading from 8.8 to 8.11 or 8.12 for example. Since SLO is GA, I'm hopeful we'll provide saved object migration every time we have to, therefore this should not happen anymore.